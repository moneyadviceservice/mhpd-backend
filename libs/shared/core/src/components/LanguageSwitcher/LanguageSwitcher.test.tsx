import React from 'react';
import { render, screen } from '@testing-library/react';
import { LanguageSwitcher } from '.';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('LanguageSwitcher component', () => {
  it('renders correctly', () => {
    render(<LanguageSwitcher testId="test-component" />);
    const container = screen.getByTestId('test-component');
    expect(container).toMatchSnapshot();
  });
});
