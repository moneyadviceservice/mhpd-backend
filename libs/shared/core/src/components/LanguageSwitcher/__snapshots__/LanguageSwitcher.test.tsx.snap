// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher component renders correctly 1`] = `
<nav
  aria-label="Language switcher"
  class="hidden md:block absolute right-[70px]"
  data-testid="test-component"
>
  <ul>
    <li
      class="inline-block after:content-[''] after:inline-block after:h-6 after:relative after:top-2 after:border-white after:border-r-1"
    >
      <span
        aria-current="true"
        class="text-white p-2 px-4"
      >
        English
      </span>
    </li>
    <li
      class="inline-block"
    >
      <a
        class="group p-2 px-4 inline-flex items-center gap-2 text-white no-underline hover:text-pink-400 visited:text-white focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 hover:underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
        href="?language=cy"
        hreflang="cy"
        lang="cy"
        rel="alternate"
      >
        <span
          class="sr-only"
        >
          Newid yr iaith ir Gymraeg
        </span>
        <span
          aria-hidden="true"
          class="inline-block relative after:content-[''] after:inline-block after:absolute after:-bottom-1 after:left-0 after:bg-white after:w-full after:h-[1px] group-hover:after:h-[2px] "
        >
          Cymraeg
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
